packages/gateway/src/features/user/usersAPI.ts

```diff
@@ -41,7 +41,8 @@ export class UsersAPI extends OpenCRVSRESTDataSource {
       return await response
     } catch (e) {
       // Don't need to throw errors if unauthorized error is found for no user with this email
-      if (e instanceof AuthenticationError) return null
+      if (e instanceof GraphQLError && e.extensions.code === 'UNAUTHENTICATED')
```

Comment:
What's the difference here to the previous code 🤔 

```ts
export class AuthenticationError extends GraphQLError {
  constructor(message = 'Unauthorized') {
    super(message, {
      extensions: {
        code: 'UNAUTHENTICATED'
      }
    })
  }
}
```

Looking at this, should this be identical in functionality?

Reply:
is it because 
```
const response = this.post('getUser', { body: { email } })
```
this calls the resolver via such a layer (using HTTP? 🙄 ) that it loses the internal JS context but the error contents with the extension flows through.

Reply:
>Looking at this, should this be identical in functionality?

That what we had thought as well when it was initially refactored. But `AuthenticationError` is a class that we are creating ourselves and not something that the library, the one throwing the error, is exposing to us. So it won't match with the thrown error

Reply:
```ts
  override didEncounterError(error: Error): never {
    if (error instanceof GraphQLError && error.extensions?.code === 'UNAUTHENTICATED') {
      throw new AuthenticationError(error.message)
    }
    throw error
  }
```

Would there be a point in doing something like this in `packages/gateway/src/graphql/data-source.ts`? So that as much of our code could trust the constructors really being `AuthenticationError` as possible?

Reply:
That is a good option